From f93ef7af9693e4c7fd2faafd9a579bd37e622d36 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 16 Nov 2016 13:58:41 -0800
Subject: [PATCH] Do not compare pointer with integer

The current code seems to be comparing a pointer to zero
which probably is not what we want. The check in current form
yields no value. Check for pointer to be non-null instead

clang says
Interpreter.cpp:813:23: error: ordered comparison between pointer and zero ('Opcode' (aka 'const void *') and 'int')
        return opcode >= 0 && static_cast<OpcodeID>(bitwise_cast<uintptr_t>(opcode)) <= op_end;
               ~~~~~~ ^  ~

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 .../JavaScriptCore/Source/JavaScriptCore/interpreter/Interpreter.cpp    | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/netflix/3rdparty/JavaScriptCore/Source/JavaScriptCore/interpreter/Interpreter.cpp b/netflix/3rdparty/JavaScriptCore/Source/JavaScriptCore/interpreter/Interpreter.cpp
index 6924ae4..1bf6c28 100644
--- a/netflix/3rdparty/JavaScriptCore/Source/JavaScriptCore/interpreter/Interpreter.cpp
+++ b/netflix/3rdparty/JavaScriptCore/Source/JavaScriptCore/interpreter/Interpreter.cpp
@@ -810,7 +810,7 @@ bool Interpreter::isOpcode(Opcode opcode)
 #if ENABLE(COMPUTED_GOTO_CLASSIC_INTERPRETER) || ENABLE(LLINT)
 #if !ENABLE(LLINT)
     if (!m_classicEnabled)
-        return opcode >= 0 && static_cast<OpcodeID>(bitwise_cast<uintptr_t>(opcode)) <= op_end;
+        return opcode != 0 && static_cast<OpcodeID>(bitwise_cast<uintptr_t>(opcode)) <= op_end;
 #endif
     return opcode != HashTraits<Opcode>::emptyValue()
         && !HashTraits<Opcode>::isDeletedValue(opcode)
-- 
2.10.2

